packages/core/src/GadgetRecord.ts

```diff
@@ -0,0 +1,230 @@
+import { klona as cloneDeep } from "klona";
+import type { Jsonify } from "type-fest";
+import { isEqual, toPrimitiveObject } from "./support.js";
+
+export enum ChangeTracking {
+  SinceLoaded,
+  SinceLastPersisted,
+}
+
+export type RecordShape = Record<string, any> | null | undefined | void;
+
+const kFields = Symbol.for("g/fields");
+const kInstantiatedFields = Symbol.for("g/if");
+const kPersistedFields = Symbol.for("g/pf");
+const kFieldKeys = Symbol.for("g/fk");
+const kTouched = Symbol.for("g/t");
+
+/** Represents one record returned from a high level Gadget API call */
+export class GadgetRecord_<Shape extends RecordShape> {
```

Comment:
Does this sucker have to be in the core package? There's a bunch of stuff I want to do to it to change it such that it'd be great to be able to generate it instead

Reply:
I think after this first refactor though we are going to be in an ok state to maybe move it over; there's going to be this new thing once it's all said and done where the generated client gives the hooks the tools it needs to do things like build operations; so I think we could extract out a "here's how to build records"